### PR TITLE
convert to babel 6

### PR DESCRIPTION
**Pull request by pwmckenna**
updated to use eslint 6. sort of a WIP

disabled babel-plugin-react-display-name...there's some [plugin ordering issue](https://github.com/gaearon/babel-plugin-react-transform/issues/19) that caused the following error 

```
ERROR in ./entry.js
Module build failed: TypeError: Plugin is not a function
    at exports.default (/Users/pwmckenna/git/react-heatpack/node_modules/babel-plugin-react-display-name/lib/index.js:57:10)
    at Function.memoisePluginContainer (/Users/pwmckenna/git/react-heatpack/node_modules/babel-core/lib/transformation/file/options/option-manager.js:127:13)
    at Function.normalisePlugin (/Users/pwmckenna/git/react-heatpack/node_modules/babel-core/lib/transformation/file/options/option-manager.js:161:32)
    at /Users/pwmckenna/git/react-heatpack/node_modules/babel-core/lib/transformation/file/options/option-manager.js:197:30
    at Array.map (native)
    at Function.normalisePlugins (/Users/pwmckenna/git/react-heatpack/node_modules/babel-core/lib/transformation/file/options/option-manager.js:173:20)
    at OptionManager.mergeOptions (/Users/pwmckenna/git/react-heatpack/node_modules/babel-core/lib/transformation/file/options/option-manager.js:271:36)
    at OptionManager.addConfig (/Users/pwmckenna/git/react-heatpack/node_modules/babel-core/lib/transformation/file/options/option-manager.js:221:10)
    at OptionManager.findConfigs (/Users/pwmckenna/git/react-heatpack/node_modules/babel-core/lib/transformation/file/options/option-manager.js:364:16)
    at OptionManager.init (/Users/pwmckenna/git/react-heatpack/node_modules/babel-core/lib/transformation/file/options/option-manager.js:412:12)
 @ multi main
```

### CODE REVIEW

**Comment by insin** [comment]
Awesome, thanks for making a start on this - very busy at the moment but will check it when I can.

Since I haven't been a good maintainer and created a `CONTRIBUTING.md` document yet, I usually test this by aliasing `heatpack` to run heatpack from my development repo and copy the `test/` folder to an independent location to ensure everything which should get resolved does get resolved.

**Comment by pwmckenna** [comment]
@insin updated and tested from a separate directory. everything looks good. one issue that I want to call out is the export of the es6-class.js file. not sure if this is specific to babel6, but using `export default` works fine if babel also compiles the `import`/`require` statements that consume it, however the compiled version compiles to something that exports a single `default` key. This means that running heatpack against es6-class.js directly doesn't work (it tries to run an object rather than the default key which is the react component), and Widget has to be updated to import it. Hopefully this is acceptable. Happy to create a CONTRIBUTING.md file as well to document some of this stuff.

**Comment by pwmckenna** [comment]
btw, happy to squash if/once its looking good

**Comment by insin** [comment]
Have this pulled down locally for review later.

Will play about with the `default` thing - wondering if Babel 6 would generate something compatible with both ES5 and ES6 default exports if we were to use ES6 imports in `entry.js` and transpile it.

File: package.json [hunk 1]
```diff
@@ -14,11 +14,15 @@
   },
   "dependencies": {
     "autoprefixer-loader": "^3.1.0",
-    "babel-core": "^5.8.34",
-    "babel-loader": "^5.4.0",
+    "babel-core": "^6.3.15",
+    "babel-loader": "^6.2.0",
     "babel-plugin-react-display-name": "^2.0.0",
```
Comment (insin) [inline package.json]: Think this should be:

``` js
"babel-plugin-transform-react-display-name": "^6.3.13",
```

File: package.json [hunk 2]
```diff
@@ -14,11 +14,15 @@
...
-    "babel-plugin-react-transform": "^1.1.1",
-    "babel-runtime": "^5.8.34",
+    "babel-plugin-react-transform": "^2.0.0-beta1",
+    "babel-plugin-transform-runtime": "^6.3.13",
+    "babel-preset-es2015": "^6.3.13",
+    "babel-preset-react": "^6.3.13",
+    "babel-preset-stage-0": "^6.3.13",
+    "babel-runtime": "^6.3.19",
```
Comment (insin) [inline package.json]: It seems like every preset plugin uses `"^5.0.0"` for this dependency - using the same version cuts out a bunch of duplication in `node_modules` - what's the deal with the plugins using v5?

**Comment by insin** [comment]
Using this to import the target module in `entry.js` seems to resolve the `default` issue, as it uses Babel's  interop require:

``` js
import Component from 'react-heatpack-script-alias'
```

File: package.json (hunk 2 above)
Comment (pwmckenna) [inline package.json]: updated...appears that all the babel 6 stuff is written in es6 and needs the babel 5 runtime to support platforms that don't support that...i believe. here's the best reference I could find : https://phabricator.babeljs.io/T6763#67858

File: package.json (hunk 1 above)
Comment (pwmckenna) [inline package.json]: updated

**Comment by insin** [comment]
Great, thanks! Will see about getting a release out this weekend

**Comment by pwmckenna** [comment]
awesome! learned a bunch doing this pr. thanks for all the feedback.

**Pull request merged**